core/src/main/java/org/elasticsearch/index/query/BoostingQueryBuilder.java

```diff
@@ -81,25 +112,91 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
         if (negativeQuery == null) {
             throw new IllegalArgumentException("boosting query requires negative query to be set");
         }
-        if (negativeBoost == -1) {
-            throw new IllegalArgumentException("boosting query requires negativeBoost to be set");
+        if (negativeBoost < 0) {
+            throw new IllegalArgumentException("boosting query requires negativeBoost to be set to positive value");
```

Comment:
not sure, I think this error should only be returned by the validate method. Here we could print out whatever we have. The fact that we print out a query doesn't mean that it's validated? That said we might have relied on validation for previous queries in doXContent, which can be a problem... thoughts?